skyportal/handlers/api/thumbnail.py

```diff
@@ -173,30 +173,30 @@ def delete(self, thumbnail_id):
         t = Thumbnail.query.get(thumbnail_id)
         if t is None:
             return self.error('Invalid thumbnail ID.')
-        # Ensure user/token has access to parent source
-        s = Source.get_if_owned_by(t.source.id, self.current_user)
+        _ = get_either_source_candidate_if_owned_by_user(t.source.id, self.current_user)
 
         DBSession.query(Thumbnail).filter(Thumbnail.id == int(thumbnail_id)).delete()
         DBSession().commit()
 
         return self.success()
 
 
-def create_thumbnail(thumbnail_data, thumbnail_type, source_id, photometry_obj):
-    basedir = Path(os.path.dirname(__file__))/'..'/'..'
+def create_thumbnail(thumbnail_data, thumbnail_type, source_id,
+                     photometry_obj):
+    basedir = Path(os.path.dirname(__file__)) / '..' / '..'
     if os.path.abspath(basedir).endswith('skyportal/skyportal'):
-        basedir = basedir/'..'
+        basedir = basedir / '..'
     file_uri = os.path.abspath(
-        basedir/f'static/thumbnails/{source_id}_{thumbnail_type}.png')
+        basedir / f'static/thumbnails/{source_id}_{thumbnail_type}.png')
     if not os.path.exists(os.path.dirname(file_uri)):
-        (basedir/'static/thumbnails').mkdir(parents=True)
+        (basedir / 'static/thumbnails').mkdir(parents=True)
     file_bytes = base64.b64decode(thumbnail_data)
     im = Image.open(io.BytesIO(file_bytes))
     if im.format != 'PNG':
         raise ValueError('Invalid thumbnail image type. Only PNG are supported.')
     if not (100, 100) <= im.size <= (500, 500):
```

Comment:
This does not do what you think it does. When the second element is outside the range, the condition will not be met. Ie.
```
(100, 100) <= (100, 2000) <= (500, 500)  == True
```

Reply:
Noted, thanks for catching that. Opening an issue and will take care of it in an upcoming PR.